### PR TITLE
Add support for Google Pub/Sub transport

### PR DESCRIPTION
**Pull request by haimjether**
* Add events support.
 * Add mingle support.
 * Add documentation

#Fixes: https://github.com/celery/kombu/issues/705
#Related: https://github.com/celery/kombu/pull/2147

### CODE REVIEW

**Review by Nusnus — COMMENTED** [review]
There's an issue with pypy. Probably not related to this PR, but it blocks the CI for now.
I'll update this branch once it's fixed so we can review it properly.

**Comment by haimjether** [comment]
I'll try to workaround it by adding an extra url at https://celery.github.io/celery-wheelhouse/repo/simple/

File: docs/getting-started/backends-and-brokers/gcpubsub.rst [hunk 1]
```diff
@@ -0,0 +1,142 @@
+.. _broker-gcpubsub:
+
+=====================
+ Using Google Pub/Sub
+=====================
```
Comment (Nusnus) [inline docs/getting-started/backends-and-brokers/gcpubsub.rst:5]:
```suggestion
=====================

.. versionadded:: 5.5
```

**Review by Nusnus — APPROVED** [review]
Added missing version to docs.
Besides, great work!
Thank you!

**Review by Nusnus — APPROVED** [review]

**Pull request merged**